### PR TITLE
Infra/unify docs

### PR DESCRIPTION
**Pull request by ethanshar**
## Description
- Create an npm package for our gatsby docs files - `uilib-docs` 
- Export all relevant configurations like plugins and page creators 

## Changelog
*Add a quick message for our users about this change*

### CODE REVIEW

File: uilib-docs/.npmignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+node_modules/
+.npmignore
+
+.cache
+public
```
Comment (M-i-k-e-l) [inline uilib-docs/.npmignore:5]: Adding a line break
```suggestion
public

```

File: uilib-docs/configurations/pageCreators/createMarkdownPages.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ module.exports = async ({graphql, boundActionCreators}) => {
   const {createPage} = boundActionCreators;
   const result = await graphql(`
     {
-      allFile(filter: {sourceInstanceName: {eq: "markdown-pages"}}) {
+      allFile(filter: {sourceInstanceName: {eq: "markdown-pages"}, childMarkdownRemark: {frontmatter: {path: {ne: null}}}}) {
```
Comment (M-i-k-e-l) [inline uilib-docs/configurations/pageCreators/createMarkdownPages.js:7]: Sure... :)
I guess we'll talk about it.

Comment (ethanshar) [reply uilib-docs/configurations/pageCreators/createMarkdownPages.js:7]: Not sure I understand the comment..

Comment (M-i-k-e-l) [reply uilib-docs/configurations/pageCreators/createMarkdownPages.js:7]: We've said we'll discuss these PRs, so let's do that when you can.

**Review by M-i-k-e-l — APPROVED** [review]

**Pull request merged**